pkg/autoupdate/tuf/library_manager.go

```diff
@@ -0,0 +1,341 @@
+package tuf
+
+import (
+	"context"
+	"crypto/sha512"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/fsutil"
+	"github.com/kolide/kit/version"
+	"github.com/kolide/launcher/ee/localserver"
+	"github.com/kolide/launcher/pkg/autoupdate"
+	client "github.com/theupdateframework/go-tuf/client"
+)
+
+// updateLibraryManager manages the update libraries for launcher and osquery.
+// It downloads and verifies new updates, and moves them to the appropriate
+// location in the library specified by the version associated with that update.
+// It also ensures that old updates are removed when they are no longer needed.
+type updateLibraryManager struct {
+	metadataClient  *client.Client // used to validate downloads
+	mirrorUrl       string         // dl.kolide.co
+	mirrorClient    *http.Client
+	rootDirectory   string
+	operatingSystem string
+	osquerier       localserver.Querier // used to query for current running osquery version
+	logger          log.Logger
+}
+
+func newUpdateLibraryManager(metadataClient *client.Client, mirrorUrl string, mirrorClient *http.Client, rootDirectory string, operatingSystem string, osquerier localserver.Querier, logger log.Logger) (*updateLibraryManager, error) {
```

Comment:
We can! I don't feel strongly about it. It's convenient to have them together because I'm reusing some test setup functions for both sets of tests, but that's not difficult to break apart either.

Reply:
You're call! I don't feel strongly

Reply:
I think I'll leave them together for now and we can always split them in the future if something changes